### DISCUSSION TITLE
fido_dev_is_fido2 on Windows Hello Pseudo-Device

### DISCUSSION THREAD

**Discussion by ohnowade** [Yubico/libfido2#736] [Q&A] [answered]
Hi there,

We are currently trying to use libfido2 on Windows. We do not require users to have administrator access so the library was built with `USE_WINHELLO=ON`. We recently ran into a problem, in which when calling `fido_dev_is_fido2` on the devices, it seemed to always return true. 

I know that libfido2 abstracts `webauthn.dll` into a `windows://hello` pseudo-device, so some APIs to access information of the devices, like `fido_dev_is_fido2`, might not work appropriately? But please correct me if I am wrong and if I am not calling this API correctly.

Also, if the above guess is true, is there any other recommended API (or workaround) to decide if a device is FIDO2 or U2F? We want to set the correct RPID if a device was registered with U2F (blind retry is the only solution on top of my head now). 

Thank you so much!

**Comment by LDVG** [comment, marked as answer]
Hi,

Yes, the return value of `fido_dev_is_fido2()` will always be true on the `windows://hello`  pseudo-device since there's no `webauthn.dll` API to query or select authenticators programatically.

> Also, if the above guess is true, is there any other recommended API (or workaround) to decide if a device is FIDO2 or U2F? We want to set the correct RPID if a device was registered with U2F (blind retry is the only solution on top of my head now).

Are you looking for [`fido_assert_set_winhello_appid()`](https://github.com/Yubico/libfido2/blob/92e3d895c30a4a1a92ecaa698ee19e08aa06c4a1/man/fido_assert_set_authdata.3#L232-L279) (which will be part of the next libfido2 release, see also https://github.com/Yubico/libfido2/pull/689)?

**Reply by ohnowade** [reply]
Hi, I think that is exactly what I was looking for. Thank you! Is there an estimated release date for it at the moment?

**Reply by LDVG** [reply]
Please refer to https://github.com/Yubico/libfido2/discussions/737 :slightly_smiling_face:

**Reply by LDVG** [reply]
Hi again @ohnowade, 1.14.0 was released 2023-11-13. :slightly_smiling_face:

**Reply by ohnowade** [reply]
Sorry for missing the previous message. Thank you so much for the update!
